src/main/java/handlers/MethodOptions2Handler.java

```diff
@@ -0,0 +1,50 @@
+package main.java.handlers;
+
+import main.java.server.HTTPMessageComponents.HTTPMethods;
+import main.java.server.response.Response;
+import main.java.server.request.Request;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import static main.java.server.HTTPMessageComponents.HTTPSyntax.*;
+import static main.java.server.HTTPMessageComponents.StatusCodes.NOT_ALLOWED;
+import static main.java.server.HTTPMessageComponents.StatusCodes.OK;
+
+public class MethodOptions2Handler implements Handler {
+
+    @Override
+    public Response generateResponseForRequest(Request request) {
+        if (!methodIsAllowed(request.getRequestMethod())) {
+            return new Response(VERSION + SP + NOT_ALLOWED + CRLF, buildHeaders());
+        }
+        return new Response(VERSION + SP + OK + CRLF, buildHeaders());
+    }
+
+    @Override
+    public String getPath() {
+        return "/method_options2";
+    }
+
+    @Override
+    public String[] allowedMethods() {
+        return new String[]{HTTPMethods.GET,
+                            HTTPMethods.HEAD,
+                            HTTPMethods.OPTIONS,
+                            HTTPMethods.PUT,
+                            HTTPMethods.POST};
+    }
+
+    private HashMap<String, String> buildHeaders() {
+        HashMap<String, String> headers = new HashMap<>();
+        String allowedHeaders = String.join(",", allowedMethods());
+        headers.put("Allow", allowedHeaders);
+        return headers;
+    }
+
+    private boolean methodIsAllowed(String method) {
+        List<String> headersList = Arrays.asList(allowedMethods());
+        return headersList.contains(method);
```

Comment:
Noticing that it seems like the above 2 functions are also potentially prone to duplication. Let's think about how to reduce the boilerplate needed to add new handlers, since I suspect we'd plan on eventually writing a lot of them.

Reply:
Hey, I know it's a bit late to fully review this, but I wanted to chime in just to pile on stuff to this specific comment.  Thinking about how to empower yourself to write a lot of handlers is 💯  IMO.

In a lot of ways, when we say "good design" we mean "make something really easy to use in a predictable way".  So one thing I'd suggest is, if you get a bit of bandwidth, take a step back and do a bit of thinking on how you'd answer that question:  "If you were writing a lot of request handlers right now, how would you _prefer_ to write them?"

More than anything else, I bet the answer to that question will give you some ideas about potential opportunities in for change already hiding in your existing handler setup.

Reply:
That's a good way to think about it, thanks! My initial thought was to promote the duplicated methods in each handler to a super and then modify behavior for inheriting handlers, but now I'm going down the path of converting the handler interface into a functional one so I can implement individualized handlers as lambdas, which would make defining new ones a breeze.